### PR TITLE
#162243954 Users should be able to register with My Diary

### PR DESCRIPTION
**Pull request by kakaemma**
#### What does this PR do?
This PR adds functionality to register users.

### How should this be manually tested?
clone the repository with `git clone`
create another branch off the current one with `git checkout -b <your_branch_name>` 
run `git pull origin ft-users-should-register`
Run `npm run mydiary` to install all dependencies and start the application
Enter the UR `http://localhost:3000/` in your browser.
click signup and use the signup page to register a user

#### What are the relevant pivotal tracker stories?
[#162243954](https://www.pivotaltracker.com/story/show/162243954)

#### Any background context you want to add?
N/A

## Checklist: I completed these to help reviewers

- [x] My IDE is configured to follow the [**ESlint Conventions**](https://github.com/eslint/eslint)

- [x] I have **added tests** to cover my changes.

- [x] All new and existing **tests passed**.

- [x] My pull request is **based on the latest changes** of the develop branch.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,20 +3,25 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
+    "D": "^1.0.0",
```
Comment (RonKbS) [inline package.json]: Do you happen to know what this is for?

File: src/tests/components/__snapshots__/LoginForm.test.js.snap [hunk 2]
```diff
@@ -104,7 +104,12 @@ ShallowWrapper {
             >
               New to My Diary ?
               <span>
-                Sign up
+                <Link
+                  replace={false}
```
Comment (samuelbwambale) [inline src/tests/components/__snapshots__/LoginForm.test.js.snap:108]: @kakaemma Consider adding the snapshots to the `gitignore` file. They are unnecessary on remote repository

**Review by Baalmart — COMMENTED** [review]
I have just run the application and the signup section is inactive. Could you be having any comments about this?

**Comment by kakaemma** [comment]
@Baalmart that is so because what you have cloned  has contents from the develop branch,  You will have to create another branch after cloning the application and run a `git pull origin ft-users-should-register`

File: src/tests/components/__snapshots__/LoginForm.test.js.snap (hunk 2 above)
Comment (kakaemma) [reply src/tests/components/__snapshots__/LoginForm.test.js.snap:108]: @samuelbwambale, according to the jest documentation https://jestjs.io/docs/en/snapshot-testing#should-snapshot-files-be-committed, snapshots should also be committed

File: package.json (hunk 1 above)
Comment (kakaemma) [reply package.json]: Thank you @RonKbS. I shall remove it

File: src/tests/components/Register.test.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    expect(wrapper).toMatchSnapshot();
+  });
+  it('should not redirect if registration is not successful', () => {
+    const nextProps = {
+      registrationSuccessful: false,
+    };
+    wrapper.setProps({ ...nextProps });
+    expect(history.push).not.toBeCalled();
+  });
+  it('should redirect if registration is successful', () => {
```
Comment (Baalmart) [inline src/tests/components/Register.test.js:44]: Try to practice positive and negative testing. In this case, you could add another tests for this scenrio like:
it('should not redirect if registration is successful')

Comment (kakaemma) [reply src/tests/components/Register.test.js:44]: Thank you @Baalmart for the feedback. However,  there is a test on line `37` that takes care of not redirecting if registration is not successful.

**Review by samuelbwambale — APPROVED** [review]
LGTM

File: src/tests/components/Register.test.js (hunk 3 above)
Comment (Baalmart) [reply src/tests/components/Register.test.js:44]: Noted Kaka, thank you.

**Pull request merged**